src/test/java/com/example/demo/OrderServiceTests.java

```diff
@@ -1,27 +1,240 @@
 package com.example.demo;
 
 import com.example.demo.matcher.models.OrderAction;
+import com.example.demo.matcher.models.OrderbookItem;
 import com.example.demo.matcher.services.OrderService;
 import com.example.demo.matcher.models.Order;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 import org.springframework.boot.test.context.SpringBootTest;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.spy;
 
 @SpringBootTest
 public class OrderServiceTests {
     OrderService orderService;
+
     @BeforeEach
     void InitialiseOrderService() {
-        orderService = new OrderService();
+        orderService = spy(new OrderService());
     }
 
     @Test
     void ItShouldAddOrders() {
         Order order1 = new Order("account1", 1, 1, OrderAction.BUY);
+        Order order2 = new Order("account1", 1, 1, OrderAction.SELL);
+
+        orderService.add(order1);
+        orderService.add(order2);
+
+        assertThat(orderService.get()).isEqualTo(Arrays.asList(order1, order2));
+    }
+
+    @Test
+    void ItShouldRemoveOrders() {
+        Order order1 = new Order("account1", 1, 1, OrderAction.BUY);
+        Order order2 = new Order("account2", 1, 1, OrderAction.SELL);
+        Order order3 = new Order("account3", 1, 1, OrderAction.SELL);
 
         orderService.add(order1);
-        assertEquals(2, 3-2);
+        orderService.add(order2);
+        orderService.add(order3);
+
+        orderService.remove(order2);
+        assertThat(orderService.get()).isEqualTo(Arrays.asList(order1, order3));
+
+        orderService.remove(order3);
+        assertThat(orderService.get()).isEqualTo(List.of(order1));
+
+        orderService.remove(order1);
+        assertThat(orderService.get()).isEqualTo(List.of());
+    }
+
+    @Test
+    void ItShouldGenerateAnOrderbookWithBuyAction() {
+        Order order1 = new Order("account1", 20, 9, OrderAction.BUY);
+        Order order2 = new Order("account2", 10, 7, OrderAction.BUY);
+        Order order3 = new Order("account3", 20, 9, OrderAction.BUY);
+        Order order4 = new Order("account4", 10, 10, OrderAction.BUY);
+        Order order5 = new Order("account5", 30, 19, OrderAction.BUY);
+        Order order6 = new Order("account6", 40, 100, OrderAction.SELL);
```

Comment:
Not a blocker - might be worth pulling the orders that are duplicated across the tests in the `@BeforeEach`

Reply:
I've put them as returns from static methods instead, since there are different ones for different groups of tests